bokeh/embed/bundle.py

```diff
@@ -420,7 +420,33 @@ def _use_mathjax(objs: Sequence[Model | Document]) -> bool:
         bool
     '''
     from ..models.text import MathText
-    return _any(objs, lambda obj: isinstance(obj, MathText))
+
+    def model_requires_mathjax(model: Model) -> bool:
+        from ..models.axes import Axis
+
+        def is_tex_string(text: str) -> bool:
+            if text.startswith("$$") and text.endswith("$$"):
+                return True
+
+            elif text.startswith("\\[") and text.endswith("\\]"):
+                return True
+
+            elif text.startswith("\\(") and text.endswith("\\)"):
+                return True
+
+            return False
+
+        if isinstance(model, Axis):
+            if isinstance(model.axis_label, str) and is_tex_string(model.axis_label):
+                return True
+
+            for val in model.major_label_overrides.values():
+                if isinstance(val, str) and is_tex_string(val):
+                    return True
```

Comment:
This is not ideal long term, since there were be several other cases to check and collecting special-cases here away from the relevant types is a pain for maintenance. But I am willing to merge this for now with an issue to improve as more math-aware locations are added.

Reply:
This is a temporary solution, much simpler and localized than everything that was proposed before. The right thing to do is to check property types, but we don't have any tooling for this.